code/lib/cli/src/autoblock/types.ts

```diff
@@ -0,0 +1,42 @@
+import type { JsPackageManager, PackageJson } from '@storybook/core-common';
+import type { StorybookConfig } from '@storybook/types';
+
+export interface AutoblockOptions {
+  packageManager: JsPackageManager;
+  packageJson: PackageJson;
+  mainConfig: StorybookConfig;
+  mainConfigPath: string;
+  configDir: string;
+}
+
+export interface Blocker<T> {
+  /**
+   * A unique string to identify the blocker with.
+   */
+  id: string;
+  /**
+   * Check if the blocker should block.
+   *
+   * @param {AutoblockOptions} options - The context.
+   * @returns {Promise<T | false>} - Return a truthy value to activate the block, return false to proceed.
```

Comment:
Since it is autogenerated, I think you haven't added the type definitions inside the JSDOC block on purpose. Since we are using TypeScript, these are redundant. Let's remove them for easier maintainability.

Reply:
What makes you think these were autogenerated? I manually crafted these so in vscode you get a bit more information what the purpose of these properties/methods. You know, for easier maintainability?

Reply:
Hmm. I still believe that JSDOC comments/descriptions shouldn't have any type information when TypeScript is used. If we take a look at the most popular eslint plugin handling JSDOC linting (https://www.npmjs.com/package/eslint-plugin-jsdoc), the
[no-types](https://github.com/gajus/eslint-plugin-jsdoc/blob/HEAD/docs/rules/no-types.md#readme)` is enabled by default for TypeScript projects, and I think as well that this makes a lot of sense. Why should we maintain type information in two places in parallel?

Reply:
Just for clarity. I am not against the description, just against using types inside the description.

Reply:
How do I change the code so that it satiesfied you quest and also you get the important information that the return value from check is passed as the second argument of message & log?

<img width="638" alt="Screenshot 2024-02-07 at 15 37 58" src="https://github.com/storybookjs/storybook/assets/3070389/d53d05cb-e17b-4578-bf27-6e7bdd556bb1">

Reply:
Is this preferred?
<img width="759" alt="Screenshot 2024-02-07 at 15 48 19" src="https://github.com/storybookjs/storybook/assets/3070389/dec8dc7f-a67b-44b0-a97a-c4b36f2f00ad">

Reply:
Is a separator like `>` needed? Without, at least in vscode I get the following codehint for this example code:

```ts
/**
 * Given a Storybook configuration object, retrieves the package name or file path of the framework.
 * @param mainConfig The main Storybook configuration object to lookup.
 * @returns The package name of the framework. If not found, returns null.
 */
export const getFrameworkPackageName = (mainConfig?: StorybookConfigRaw) => {
```

![image](https://github.com/storybookjs/storybook/assets/5889929/43adedb9-b97c-4953-b048-4594940be912)

Reply:
But for readability, we can just use it. I don't have anything against it.

Reply:
Here's why I did it:
<img width="553" alt="Screenshot 2024-02-08 at 00 13 25" src="https://github.com/storybookjs/storybook/assets/3070389/5a7fd601-e571-4a42-b328-b301a89f0085">
<img width="565" alt="Screenshot 2024-02-08 at 00 13 33" src="https://github.com/storybookjs/storybook/assets/3070389/b9c8b652-2cc8-4b47-b24b-290552843b87">

Reply:
The code example you've given seems a completely different bit of code, I don't see how it's relevant?

Reply:
I gave it another iteration... It feels very bike-shaddy though... All I want is it to be obvious that the return value of check is available as the second argument in both `message` and `log`.

I think the types don't make it really obvious that that's the intent, so I figured I'd make it as obvious as possible and the only way I know how to do this was with JSdoc comments.

I don't know what else to do.